test/functional/p2p_getaddr_caching.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test addr response caching
+"""
+
+from test_framework.messages import (
+    CAddress,
+    NODE_NETWORK,
+    NODE_WITNESS,
+    msg_addr,
+    msg_getaddr,
+)
+from test_framework.mininode import (
+    P2PInterface,
+    mininode_lock
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    wait_until
+)
+import time
+
+ADDRS = []
+for i in range(1000):
+    addr = CAddress()
+    addr.time = int(time.time())
+    addr.nServices = NODE_NETWORK | NODE_WITNESS
+    third_octet = int(i / 256) % 256
```

Comment:
prefer `(i >> 8) % 256` to avoid having to convert between ints and floats (and better signal that you're removing the bottom 8 bits for the third section of the IP address). Better yet: `i >> 8` since your loop doesn't go over 256^2.